packages/ember-data/tests/integration/serializers/embedded-records-mixin-test.js

```diff
@@ -1157,6 +1157,46 @@ test("serialize with embedded object (belongsTo relationship)", function() {
   });
 });
 
+test("serialize with embedded object (polymorphic belongsTo relationship)", function() {
+  env.registry.register('serializer:super-villain', DS.RESTSerializer.extend(DS.EmbeddedRecordsMixin, {
+    attrs: {
+      secretLab: { embedded: 'always' }
+    }
+  }));
+
+  SuperVillain.reopen({
+    secretLab: DS.belongsTo('secret-lab', { polymorphic: true })
+  });
+
+  var json, tom;
+  run(function() {
+    tom = env.store.createRecord(
+      'super-villain',
+      { firstName: "Tom", lastName: "Dale", id: "1",
+        secretLab: env.store.createRecord('bat-cave', { minionCapacity: 5000, vicinity: "California, USA", id: "101", infiltrated: true }),
+        homePlanet: env.store.createRecord('home-planet', { name: "Villain League", id: "123" })
+      }
+    );
+  });
+
+  run(function() {
+    json = tom.serialize();
+  });
+
+  deepEqual(json, {
+    firstName: get(tom, "firstName"),
+    lastName: get(tom, "lastName"),
+    homePlanet: get(tom, "homePlanet").get("id"),
+    secretLabType: 'batCave',
+    secretLab: {
+      id: get(tom, "secretLab").get("id"),
+      minionCapacity: get(tom, "secretLab").get("minionCapacity"),
+      vicinity: get(tom, "secretLab").get("vicinity"),
+      infiltrated: true
+    }
+  });
```

Comment:
I am not sure if this is the way this should work, or if the `type` of the embedded, polymorphic record should be a property within the `secretLab` object ...